server/src/main/java/org/elasticsearch/action/support/broadcast/unpromotable/TransportBroadcastUnpromotableAction.java

```diff
@@ -76,6 +76,17 @@ protected TransportBroadcastUnpromotableAction(
 
     @Override
     protected void doExecute(Task task, Request request, ActionListener<Response> listener) {
+        beforeDispatchingRequestToUnpromotableShards(
```

Comment:
nit: maybe add this in `TransportUnpromotableShardRefreshAction` directly, unless it is needed for tests?